pkg/jobmanager/jobmanager.go

```diff
@@ -100,6 +100,15 @@ func New(l api.Listener, pr *pluginregistry.PluginRegistry, opts ...Option) (*Jo
 }
 
 func (jm *JobManager) handleEvent(ctx xcontext.Context, ev *api.Event) {
+	// handle status asynchronously, keep rest sync
```

Comment:
i don't think we need to be making the distinction. code is already written with the assumption that multiple requests can be in flight at the same time, that's why we have jobsMu for example.

so, i'd say `go jm.handleEvent' in Run is all we need.

however, we also need a way to drain all requests gracefully when stopping the API.

Reply:
I was considering doing just `go jm.handleEvent` but didn't want to risk turning enabling concurrency everywhere at once.

Re: Draining. The context will abort in-flight requests, right? Do you think we need a way to wait for all of them to finish?

Reply:
> I was considering doing just `go jm.handleEvent` but didn't want to risk turning enabling concurrency everywhere at once.

i think there isn't much risk, i certainly was already under assumption that things can go in parallel. `JobManager.jobs` is the only bit of shared state and we already have `jobsMu` for that.

> The context will abort in-flight requests, right? Do you think we need a way to wait for all of them to finish?

We should still wait for all requests to properly fold before returning. A simple waitgroup should do.

Reply:
Ok everything is async now, and I tested the waiting with a long sleep in status (note the time jump).
```
^C[2021-04-22T16:27:41.711+01:00 D httplistener.go:235] Received server shut down request
[2021-04-22T16:27:41.711+01:00 I server.go:215] Signal "interrupt", canceling
[2021-04-22T16:27:41.711+01:00 I jobmanager.go:157] Listener shut down successfully.
[2021-04-22T16:27:59.213+01:00 I server.go:224] Exiting, <nil>
```

Want me to add messages ("Waiting for handler shutdown")?